upup/models/cloudup/resources/addons/networking.projectcalico.org/pre-k8s-1.6.yaml.template

```diff
@@ -199,3 +199,53 @@ spec:
             - name: CONFIGURE_ETC_HOSTS
               value: "true"
 
+{{ if and (eq .CloudProvider "aws") (.Networking.Calico.CrossSubnet) -}}
+---
+# This manifest installs the k8s-ec2-srcdst container, which disables
+# src/dst ip checks to allow BGP to function for calico for hosts within subnets
+# This only applies for AWS environments.
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: k8s-ec2-srcdst
+  namespace: kube-system
+  labels:
+    k8s-app: k8s-ec2-srcdst
+    role.kubernetes.io/networking: "1"
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      k8s-app: k8s-ec2-srcdst
+  template:
+    metadata:
+      labels:
+        k8s-app: k8s-ec2-srcdst
+        role.kubernetes.io/networking: "1"
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ''
+        scheduler.alpha.kubernetes.io/tolerations: |
+          [{"key": "dedicated", "value": "master", "effect": "NoSchedule" },
+           {"key":"CriticalAddonsOnly", "operator":"Exists"}]
+    spec:
+      containers:
+        - image: ottoyiu/k8s-ec2-srcdst:v0.0.3
+          name: k8s-ec2-srcdst
+          resources:
```

Comment:
I would rather have it set, since you know the app best. We should probably file an issue to ensure that all cni providers have memory set.